scripts/effects/hagakure.lua

```diff
@@ -7,8 +7,8 @@ local effectObject = {}
 effectObject.onEffectGain = function(target, effect)
     local jpValue = target:getJobPointLevel(xi.jp.HAGAKURE_EFFECT)
 
-    target:addMod(xi.mod.SAVETP, 400)
-    target:addMod(xi.mod.TP_BONUS, 1000 + (jpValue * 10))
+    target:addMod(xi.mod.SAVETP, effect:getPower())
+    target:addMod(xi.mod.TP_BONUS, effect:getSubPower() + (jpValue * 10))
```

Comment:
```lua
    effect:addMod(xi.mod.SAVETP, effect:getPower())
    effect:addMod(xi.mod.TP_BONUS, effect:getSubPower() + jpValue * 10)
```
Move modifiers from target to the effect. This means deleting onEffectLoose logic.
Also, don't use parenthesis when they aren't needed.